lib/vmpooler/api/v1.rb

```diff
@@ -273,7 +305,9 @@ def need_token!
               backend.hset('vmpooler__active__' + key, vm, Time.now)
               backend.hset('vmpooler__vm__' + vm, 'checkout', Time.now)
 
-              if Vmpooler::API.settings.config[:auth] and has_valid_token?
+              if Vmpooler::API.settings.config[:auth] and has_token?
+                validate_token(backend)
```

Comment:
Note that I'm intentionally calling `validate_token(backend)` here (and below at line 386) instead of the `need_token!` shortcut as I thought it made it more readable.